src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -253,22 +254,24 @@ public boolean logOut(boolean showPrompt) {
     return true;
   }
 
-  private Credential makeCredential(String accessToken, String refreshToken) {
-    Credential cred =
-        new GoogleCredential.Builder()
-            .setJsonFactory(jsonFactory)
-            .setTransport(transport)
-            .setClientSecrets(clientId, clientSecret)
-            .build();
-    cred.setAccessToken(accessToken);
-    cred.setRefreshToken(refreshToken);
-    return cred;
+  private Credential makeCredential(
+      String accessToken, String refreshToken, long expiresInSeconds) {
+    Credential credential = new GoogleCredential.Builder()
+        .setJsonFactory(jsonFactory)
+        .setTransport(transport)
+        .setClientSecrets(clientId, clientSecret)
+        .build();
+    credential.setAccessToken(accessToken)
+              .setRefreshToken(refreshToken)
+              .setExpiresInSeconds(expiresInSeconds);
+    return credential;
   }
 
   private void updateLoginState(GoogleTokenResponse tokenResponse)
       throws IOException, EmailAddressNotReturnedException {
     Credential credential = makeCredential(tokenResponse.getAccessToken(),
```

Comment:
I think you might be able to simplify this with `new GoogleCredential().setFromTokenResponse(tokenResponse);`